phoenix-hive/src/main/java/org/apache/phoenix/hive/PhoenixStorageHandler.java

```diff
@@ -0,0 +1,147 @@
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.metastore.HiveMetaHook;
+import org.apache.hadoop.hive.ql.metadata.DefaultStorageHandler;
+import org.apache.hadoop.hive.ql.metadata.HiveStoragePredicateHandler;
+import org.apache.hadoop.hive.ql.plan.ExprNodeDesc;
+import org.apache.hadoop.hive.ql.plan.TableDesc;
+import org.apache.hadoop.hive.serde.Constants;
+import org.apache.hadoop.hive.serde.serdeConstants;
+import org.apache.hadoop.hive.serde2.Deserializer;
+import org.apache.hadoop.hive.serde2.SerDe;
+import org.apache.hadoop.hive.serde2.SerDeException;
+import org.apache.hadoop.mapred.InputFormat;
+import org.apache.hadoop.mapred.JobConf;
+import org.apache.hadoop.mapred.OutputFormat;
+import org.apache.phoenix.hive.util.HiveConfigurationUtil;
+
+
+/**
+* PhoenixStorageHandler
+* This class manages all the Phoenix/Hive table initial configurations and SerDe Election
+*/
+
+public class PhoenixStorageHandler extends DefaultStorageHandler implements
+        HiveStoragePredicateHandler {
+    static Log LOG = LogFactory.getLog(PhoenixStorageHandler.class.getName());
+
+    private Configuration conf = null;
+
+    public PhoenixStorageHandler() {
+    }
+
+    @Override
+    public Configuration getConf() {
+        return conf;
+    }
+
+    @Override
+    public void setConf(Configuration conf) {
+        this.conf = conf;
+    }
+
+    @Override
+    public HiveMetaHook getMetaHook() {
+        return new PhoenixMetaHook();
+    }
+
+    @Override
+    public void configureInputJobProperties(TableDesc tableDesc, Map<String, String> jobProperties) {
+        configureJobProperties(tableDesc, jobProperties);
+    }
+
+    @Override
+    public void
+            configureOutputJobProperties(TableDesc tableDesc, Map<String, String> jobProperties) {
+        configureJobProperties(tableDesc, jobProperties);
+    }
+
+    @Override
+    public void configureTableJobProperties(TableDesc tableDesc, Map<String, String> jobProperties) {
+        configureJobProperties(tableDesc, jobProperties);
+    }
+
+    /**
+     * Extract all job properties to configure this job 
+     * parameter tableDesc tabledescription, jobProperties
+     * TODO this avoids any pushdown must revisit
+     */
+    private void configureJobProperties(TableDesc tableDesc, Map<String, String> jobProperties)
+    {
+      Properties tblProps = tableDesc.getProperties();
+      tblProps.getProperty("phoenix.hbase.table.name");
+      HiveConfigurationUtil.setProperties(tblProps, jobProperties);
+
+      //TODO this avoids any pushdown must revisit and extract meaningful parts
+      jobProperties.put("phoenix.select.stmt", "select * from " + (String)jobProperties.get("phoenix.hbase.table.name"));
+      if(((String)jobProperties.get("phoenix.hbase.table.name")).contains("limit")==true){
+          
+      }
+      LOG.debug("ConfigurationUtil.SELECT_STATEMENT " + (String)jobProperties.get("phoenix.select.stmt"));
+    }
+    
+    /**
```

Comment:
Can you provide an implementation of HiveStorageHandler#configureJobConf and have HBaseConfiguration passed to the JobConf. This way, the InputFormat classes and RecordReaders have the necessary HBase configurations.